rpcs3/Emu/RSX/GSRender.h

```diff
@@ -57,9 +57,14 @@ using display_handle_t = HWND;
 using display_handle_t = void*; // NSView
 #else
 using display_handle_t = std::variant<
+#ifdef HAVE_X11
 	std::pair<Display*, Window>
+#endif
+#if defined(HAVE_X11) && defined(VK_USE_PLATFORM_WAYLAND_KHR)
```

Comment:
Move to the block above or below

Reply:
I prefer below in this case.

Reply:
The issue is you can’t declare a `std::variant<Something, SomethingElse,>` type, with the extra comma at the end. :/

I’ve left it like that for now, otherwise it can’t compile when Wayland isn’t available, or when X11 isn’t available.

Reply:
I mean
```
#if X11
start decl
#endif

#if Wayland
#if X11
,
#endif
other wayland stuff
#endif
```

Reply:
Another option is to just stop with the code stitching and have explicit blocks:
```
#if defined X11 && defined Wayland
... declare full struct
#elif defined X11
... declare partial for X11 only
#else
... declare partial for wayland only
#endif
```
I feel this would make the code readable again without adding any extra loc.